src/main/java/nl/tudelft/ti2206/group9/level/save/Parser.java

```diff
@@ -127,11 +132,33 @@ private static void parseJSON(final JSONObject mainObject) {
         final JSONObject highObj = (JSONObject) mainObject.get("highscore");
         highScore = (Long) highObj.get("score");
 
-        andy = (boolean) mainObject.get("andy");
-        boy = (boolean) mainObject.get("boy");
-        captain = (boolean) mainObject.get("captain");
-        iron = (boolean) mainObject.get("iron");
-        plank = (boolean) mainObject.get("plank");
+        parseJSONShopItems(mainObject);
+    }
+
+    /**
+     * Parses the JSON data with respect to shopItems.
+     * @param mainObject the main JSON object.
+     */
+    @SuppressWarnings("rawtypes")
+    // SuppressWarnings is needed for casting the mainObject when
+    // getting the shopItems.
+    private static void parseJSONShopItems(final Object mainObject) {
```

Comment:
Couldn't you better pass the `shopItems` object instead of the entire main? I think `shopItems` is all you need here.

<logic, review